vim25/xml/util/util.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package util implements some helper functions for marshaling/unmarshaling
+// vim objects to/from XML.
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/vim25/xml"
+)
+
+// VimObjectsFromBytes returns all vim objects decoded from the provided bytes.
+//
+// This is a helper function that wraps VimObjectsFromDecoder. Please see
+// the documentation on VimObjectsFromDecoder for additional information.
+func VimObjectsFromBytes(v []byte) (<-chan interface{}, <-chan error) {
+	return VimObjectsFromReader(bytes.NewReader(v))
+}
+
+// VimObjectsFromString returns all vim objects decoded from the provided
+// string.
+//
+// This is a helper function that wraps VimObjectsFromDecoder. Please see
+// the documentation on VimObjectsFromDecoder for additional information.
+func VimObjectsFromString(v string) (<-chan interface{}, <-chan error) {
+	return VimObjectsFromReader(strings.NewReader(v))
+}
+
+// VimObjectsFromReader returns all vim objects decoded from the provided input
+// stream.
+//
+// This is a helper function that wraps VimObjectsFromDecoder. Please see
+// the documentation on VimObjectsFromDecoder for additional information.
+func VimObjectsFromReader(r io.Reader) (<-chan interface{}, <-chan error) {
+	return VimObjectsFromDecoder(xml.NewDecoder(r))
+}
+
+// VimObjectsFromDecoder returns all vim objects from the provided decoder.
+//
+// Any decode errors are sent on the error channel.
+//
+// Both channels are closed when io.EOF is reached.
```

Comment:
nit: there's another err scenario when both channels are closed. This should be documented.